### PR TITLE
Function to generate train, test and/or validation set

### LINKED ISSUE #1498: Is there a function to generate a train set and test set from DataFrame？
I find sandbox>tool has cross_validate.py, and there are many functions from sklearn include split , are there any update?

### PR DESCRIPTION
**Pull request by NolanMP**
- [ ] closes #1498
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>
 Function to generate train and test set from data frame was needed.

Added temporary directory statsmodels/statsmodels/datasplit

Added function with a dual feature in file statsmodels/statsmodels/datasplit/datasplitter.py
function: train_test_split()
1. Train test split
2. Train test validation split

Supports Pandas, Lists, and Arrays
Supports shuffle and stratify

Added test cases in file statsmodels/statsmodels/datasplit/testcases.py

</details>

### CODE REVIEW

**Comment by bashtage** [comment]
I'm not sure statsmodels is the right place for this.  Have you proposed this in sklearn-pandas?

**Comment by NolanMP** [comment]
Hi, @bashtage Thank you for your reply. I will propose this to sklearn and pandas as you suggested and update this thread later.
